.github/workflows/release.yml

```diff
@@ -1,90 +1,70 @@
+        os: [ubuntu-20.04, ubuntu-22.04]
+
+    runs-on: ["${{ matrix.os }}"]
+
     steps:
-      - uses: actions/checkout@v2
-      - name: Set env vars
-        run: |
-          source ci/env.sh
-          echo "GEYSER_PLUGIN_NAME=$plugin_name" | tee -a $GITHUB_ENV
-          echo "GEYSER_PLUGIN_LIB=lib${plugin_lib_name}" | tee -a $GITHUB_ENV
-      - if: runner.os == 'Linux'
-        run: |
-          sudo apt-get update
-          sudo apt-get upgrade
-          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
-          sudo apt-add-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-10 main"
-          sudo apt-get update
-          sudo apt-get install -y libudev-dev libssl-dev libsasl2-dev libzstd-dev
-          sudo apt-get install -y openssl --allow-unauthenticated
-          sudo apt-get install -y libssl1.1 --allow-unauthenticated
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: ${{ env.RUST_VERSION }}
-          override: true
-          profile: minimal
-          components: rustfmt
-      - name: Build Plugin
-        run: |
-          echo "CI_TAG=${GITHUB_REF#refs/*/}" >> "$GITHUB_ENV"
-          echo "CI_OS_NAME=linux" >> "$GITHUB_ENV"
-          cargo build --release
-      - name: Build release tarball
-        run: ./ci/create-tarball.sh
-      - name: Publish to crates registry
-        run: |
-          cargo publish -p plerkle_serialization --token $CARGO_TOKEN --no-verify || true
-          sleep 30
-          cargo publish -p plerkle_messenger --token $CARGO_TOKEN --no-verify || true
-        shell: bash
-        env:
-          CARGO_TOKEN: ${{ secrets.CARGO_TOKEN }}
-      - name: Release
-        uses: softprops/action-gh-release@v1
-        if: startsWith(github.ref, 'refs/tags/')
-        with:
-          name: STABLE ${{ env.CI_TAG }} 
-          body: |
-            ## STABLE VERSION: 
-            Reccomended for Mainnet
-            ${{ env.GEYSER_PLUGIN_NAME }} ${{ env.CI_TAG }}
-            solana ${{ env.SOLANA_VERSION_STABLE }}
+      
+    - name: checkout repo
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+        lfs: true
+
+    - name: set env vars
+      run: |
+        source ci/env.sh
+        echo "RUST_VERSION=$rust_version" | tee -a "$GITHUB_ENV"
+        echo "RUST_PROFILE=$rust_profile" | tee -a "$GITHUB_ENV"
+        echo "RUST_COMPONENTS=$rust_components" | tee -a "$GITHUB_ENV"
+        echo "SOLANA_VERSION=$solana_version" | tee -a "$GITHUB_ENV"
+        echo "PLUGIN_VERSION=$plugin_version" | tee -a "$GITHUB_ENV"
+        echo "RELEASE=$release" | tee -a "$GITHUB_ENV"
+
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ env.RUST_VERSION }}
+        override: true
+        profile: ${{ env.RUST_PROFILE }}
+        components: ${{ env.RUST_COMPONENTS }}
+
+    - name: Build plerkle plugin
+      run: cargo build --verbose --release
+
+    - name: Package libplerkle
```

Comment:
This seems to run for both OS, while the next one only runs for `ubuntu-22.04`. Wonder if this is missing an `if` statement. Also, related to the "matrix" strategy, we might need just one of these, since it will run the whole job twice.

Reply:
The idea of the job running in a matrix, is that it will execute the same code in each runner, but additionally, you can add task that will execute with the if statement, in a specific runner. 
If you want to tun the job only once, you need to define a single os version for the release. 
Or you can have different releases in different jobs, or even files. 
the way it's implemented here you have less duplicated code.